src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -357,15 +359,40 @@ void SetArrayItemNode(object d)
                         }
                         else if (node is ValueExecutionNode valueNode)
                         {
-                            node.Result = valueNode.GraphType.Serialize(d)
-                                ?? throw new InvalidOperationException($"Unable to serialize '{d}' to '{valueNode.GraphType.Name}' for list index {index}.");
+                            bool defaultException = false;
+                            try
+                            {
+                                object result = valueNode.GraphType.Serialize(d);
+                                if (result == null)
+                                {
+                                    defaultException = true;
+                                    if (listType.ResolvedType is NonNullGraphType)
+                                        throw new InvalidOperationException($"Unable to serialize '{d}' to '{valueNode.GraphType.Name}' for list index {index}.");
+                                    else
+                                        throw new ExecutionError($"Unable to serialize '{d}' to '{valueNode.GraphType.Name}'.");
+                                }
+                                node.Result = result;
+                            }
+                            catch (Exception error) when (listType.ResolvedType is NonNullGraphType)
```

Comment:
This should first catch `ObjectDisposedExceptions` when cancellation is requested, and throw immediately.  `ThrowIfCancellationRequested` is not a suitable replacement.

For example, if you wrote graphql code to call a REST API, and your call to graphql was hanging, and you pressed "stop" in your browser or similar to stop the call, you would want the full stack trace returned to the caller.  It should not be capturing and rethrowing any other type of exception, or returning a failed execution document.  It should throw with the original stack trace so it can be seen where the execution died.

Reply:
Actually, it's impossible for Serialize to throw (due to the cancellation token) because it's not an async call.  So nevermind.

Reply:
> it's impossible for Serialize to throw 

I spoke about it.